src/main/java/homework/task_16_multithreading_advanced/Main.java

```diff
@@ -0,0 +1,34 @@
+package homework.task_16_multithreading_advanced;
+
+public class Main {
+    public static void main(String[] args) {
+
+        PetrolStation petrolStation = new PetrolStation(100);
+
+        Thread car1 = new Thread(()-> petrolStation.doRefuel(30));
+
+        Thread car2 = new Thread(()-> petrolStation.doRefuel(20));
+
```

Comment:
Please, use ExecutorService with predefined pool size and use submit to start thread logic

Reply:
corrected